internal/eval/eval.go

```diff
@@ -693,6 +689,88 @@ func evalJSONGET(args []string, store *dstore.Store) []byte {
 	return clientio.Encode(string(resultBytes), false)
 }
 
+// evalJSONTOGGLE toggles a boolean value stored at the specified key and path.
+// args must contain at least the key and path (where the boolean is located).
+// If the key does not exist or is expired, it returns response.RespNIL.
+// If the field at the specified path is not a boolean, it returns an encoded error response.
+// If the boolean is `true`, it toggles to `false` (returns :0), and if `false`, it toggles to `true` (returns :1).
+// Returns an encoded error response if the incorrect number of arguments is provided.
+func evalJSONTOGGLE(args []string, store *dstore.Store) []byte {
+    if len(args) < 2 {
+        return diceerrors.NewErrArity("JSON.TOGGLE")
+    }
+    key := args[0]
+    path := args[1]
+
+    obj := store.Get(key)
+    if obj == nil {
+        return diceerrors.NewErrWithFormattedMessage("-ERR could not perform this operation on a key that doesn't exist")
+    }
+
+    errWithMessage := object.AssertTypeAndEncoding(obj.TypeEncoding, object.ObjTypeJSON, object.ObjEncodingJSON)
+    if errWithMessage != nil {
+        return errWithMessage
+    }
+
+    jsonData := obj.Value
+    expr, err := jp.ParseString(path)
+    if err != nil {
+        return diceerrors.NewErrWithMessage("invalid JSONPath")
+    }
+
+    toggleResults := []interface{}{}
+    modified := false
+
+    _, err = expr.Modify(jsonData, func(value interface{}) (interface{}, bool) {
+        if boolValue, ok := value.(bool); ok {
+            newValue := !boolValue
+            toggleResults = append(toggleResults, boolToInt(newValue))
+            modified = true
+            return newValue, true
+        }
+        toggleResults = append(toggleResults, nil)
+        return value, false
```

Comment:
nit: You can improve readability by making the smaller if-blocks.
```suggestion
        boolValue, ok := value.(bool)
        if !ok {
            toggleResults = append(toggleResults, nil)
            return value, false
        }
        newValue := !boolValue
        toggleResults = append(toggleResults, boolToInt(newValue))
        modified = true
        return newValue, true
```